pkg/mdformatter/linktransformer/link.go

```diff
@@ -175,7 +178,30 @@ func NewValidator(logger log.Logger, linksValidateConfig []byte, anchorDir strin
 	v.c.OnError(func(response *colly.Response, err error) {
 		v.rMu.Lock()
 		defer v.rMu.Unlock()
-		v.remoteLinks[response.Ctx.Get(originalURLKey)] = errors.Wrapf(err, "%q not accessible; status code %v", response.Request.URL.String(), response.StatusCode)
+		retriesStr := response.Ctx.Get(retryKey)
+		retries, _ := strconv.Atoi(retriesStr)
+		switch response.StatusCode {
+		case 429:
+			if retries <= 2 {
+				response.Ctx.Put(retryKey, strconv.Itoa(retries+1))
+				retryAfter, convErr := strconv.Atoi(response.Headers.Get("Retry-After"))
+				if convErr == nil {
+					time.Sleep(time.Duration(retryAfter) * time.Second)
```

Comment:
hmmmmm I think we need to have global retry. You see that will not work if others for same host will try in the same time 🤔

Reply:
Oh! I didn't think of this before! Will try to implement global retry!

Reply:
I thought about this and tested it a bit. I think we never make requests to the same host again or at the same time due to our present caching layer and retries can only occur after request! So this should work. 🙂

Reply:
I challenge this statement (: 
![](https://media3.giphy.com/media/PMpYZ0Fey755AEY5J0/giphy-downsized-medium.gif)

Reply:
I think it's fine for now, let's iterate (global retry will  be hard to implement), but I think it's needed, paths are cached not hosts